### PR TITLE
feat(3rd party board): Add Fri3d Badge 2024 (ESP32-S3-WROOM-1)

### PR DESCRIPTION
**Pull request by francisduvivier**
## Description of Change
Add Fri3d Badge 2024 (ESP32-S3-WROOM-1) as a new 3rd party board

## Tests scenarios
I have tested my PR on the Fri3d Badge 2024 (ESP32-S3-WROOM-1) 

## Related links
Board Image: https://github.com/Fri3dCamp/badge_2024/blob/main/docs/badge2024.jpg
More info on https://github.com/Fri3dCamp/badge_2024

### CODE REVIEW

**Comment by francisduvivier** [comment]
If I want to get this board in the v2.0.x release, should I then also make a PR towards the `release/v2.0.x` branch?

File: variants/fri3d_2024_esp32s3/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+static const uint8_t LED_BUILTIN = 21;
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
+#define LED_BUILTIN LED_BUILTIN
+#define RGB_BUILTIN LED_BUILTIN
+#define RGB_BRIGHTNESS 64
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<49)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 46)
```
Comment (P-R-O-C-H-Y) [inline variants/fri3d_2024_esp32s3/pins_arduino.h:22]: This part made been moved from pins_arduino.h file, and is defined globally per SoC.
Please remove it.
```suggestion
```

File: variants/fri3d_2024_esp32s3/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define USB_VID 0x303a
+#define USB_PID 0x1001
+
+#define EXTERNAL_NUM_INTERRUPTS 46
+#define NUM_DIGITAL_PINS        48
+#define NUM_ANALOG_INPUTS       20
```
Comment (P-R-O-C-H-Y) [inline variants/fri3d_2024_esp32s3/pins_arduino.h:12]: This part made been moved from pins_arduino.h file, and is defined globally per SoC.
Please remove it.
```suggestion
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@francisduvivier PTAL on my comments.

**Comment by P-R-O-C-H-Y** [comment]
> If I want to get this board in the v2.0.x release, should I then also make a PR towards the `release/v2.0.x` branch?

If you want to get the board to the 2.x branch, you can open the PR targeting that branch. But it's not recommended anymore as IDF 4.4 is EOL and we don't have any plans to release a new 2.X version.

File: boards.txt [hunk 3]
```diff
@@ -26494,6 +26494,233 @@ fm-devkit.menu.EraseFlash.none.upload.erase_cmd=
...
+fri3d_2024_esp32s3.serial.disableDTR=false
+fri3d_2024_esp32s3.serial.disableRTS=false
+
+fri3d_2024_esp32s3.build.tarch=xtensa
+fri3d_2024_esp32s3.build.bootloader_addr=0x0
+fri3d_2024_esp32s3.build.target=esp32s3
+fri3d_2024_esp32s3.build.mcu=esp32s3
+fri3d_2024_esp32s3.build.core=esp32
+fri3d_2024_esp32s3.build.variant=fri3d_2024_esp32s3
+fri3d_2024_esp32s3.build.board=FRI3D-BADGE-2024_ESP32S3-WROOM-1
```
Comment (P-R-O-C-H-Y) [inline boards.txt:26527]: One more thing, - (dash) is not valid character for build.board name. Can you please fix it as suggestion?
```suggestion
fri3d_2024_esp32s3.build.board=FRI3D_BADGE_2024_ESP32S3_WROOM_1
```
Or maybe do it simpler by removing the WROOM_1, the esp32s3 can remain :)

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
One last change needed :)

File: boards.txt (hunk 3 above)
Comment (francisduvivier) [reply boards.txt:26527]: Thanks a lot for checking this PR and the suggestions!
I change this and then also still moved the order of options for better defaults. But I think all should be good now

File: boards.txt [hunk 4]
```diff
@@ -26494,6 +26494,229 @@ fm-devkit.menu.EraseFlash.none.upload.erase_cmd=
...
+
+fri3d_2024_esp32s3.menu.PSRAM.default=OPI PSRAM
+fri3d_2024_esp32s3.menu.PSRAM.default.build.defines=-DBOARD_HAS_PSRAM
+fri3d_2024_esp32s3.menu.PSRAM.default.build.psram_type=opi
+fri3d_2024_esp32s3.menu.PSRAM.disabled=Disabled
+fri3d_2024_esp32s3.menu.PSRAM.disabled.build.defines=
+fri3d_2024_esp32s3.menu.PSRAM.disabled.build.psram_type=qspi
+fri3d_2024_esp32s3.menu.PSRAM.enabled=QSPI PSRAM
+fri3d_2024_esp32s3.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM
+fri3d_2024_esp32s3.menu.PSRAM.enabled.build.psram_type=qspi
```
Comment (P-R-O-C-H-Y) [inline boards.txt:26567]: Not possible option on `ESP32-S3-WROOM-1-N16R8`, PSRAM works only on OPI mode.
```suggestion
```

File: boards.txt [hunk 5]
```diff
@@ -26494,6 +26494,229 @@ fm-devkit.menu.EraseFlash.none.upload.erase_cmd=
...
+fri3d_2024_esp32s3.build.flash_size=16MB
+fri3d_2024_esp32s3.build.flash_freq=80m
+fri3d_2024_esp32s3.build.flash_mode=dio
+fri3d_2024_esp32s3.build.boot=qio
+fri3d_2024_esp32s3.build.boot_freq=80m
+fri3d_2024_esp32s3.build.partitions=default
+fri3d_2024_esp32s3.build.defines=
+fri3d_2024_esp32s3.build.loop_core=
+fri3d_2024_esp32s3.build.event_core=
+fri3d_2024_esp32s3.build.psram_type=qspi
```
Comment (P-R-O-C-H-Y) [inline boards.txt:26543]: Change to OPI, as qspi is not possible on `ESP32-S3-WROOM-1-N16R8`
```suggestion
fri3d_2024_esp32s3.build.psram_type=opi
```

File: boards.txt [hunk 6]
```diff
@@ -26494,6 +26494,229 @@ fm-devkit.menu.EraseFlash.none.upload.erase_cmd=
...
+fri3d_2024_esp32s3.menu.FlashMode.opi.build.boot_freq=80m
+fri3d_2024_esp32s3.menu.FlashMode.opi.build.flash_freq=80m
+
+fri3d_2024_esp32s3.menu.FlashSize.default=16MB (128Mb)
+fri3d_2024_esp32s3.menu.FlashSize.default.build.flash_size=16MB
+fri3d_2024_esp32s3.menu.FlashSize.4M=4MB (32Mb)
+fri3d_2024_esp32s3.menu.FlashSize.4M.build.flash_size=4MB
+fri3d_2024_esp32s3.menu.FlashSize.8M=8MB (64Mb)
+fri3d_2024_esp32s3.menu.FlashSize.8M.build.flash_size=8MB
+fri3d_2024_esp32s3.menu.FlashSize.8M.build.partitions=default_8MB
```
Comment (P-R-O-C-H-Y) [inline boards.txt:26597]: Can be probably removed, as your badge is using the `ESP32-S3-WROOM-1-N16R8` with 16MB flash.
```suggestion
```

File: boards.txt [hunk 7]
```diff
@@ -26494,6 +26494,229 @@ fm-devkit.menu.EraseFlash.none.upload.erase_cmd=
...
+fri3d_2024_esp32s3.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
+fri3d_2024_esp32s3.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FATFS)
+fri3d_2024_esp32s3.menu.PartitionScheme.fatflash.build.partitions=ffat
+fri3d_2024_esp32s3.menu.PartitionScheme.fatflash.upload.maximum_size=2097152
+fri3d_2024_esp32s3.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9.9MB FATFS)
+fri3d_2024_esp32s3.menu.PartitionScheme.app3M_fat9M_16MB.build.partitions=app3M_fat9M_16MB
+fri3d_2024_esp32s3.menu.PartitionScheme.app3M_fat9M_16MB.upload.maximum_size=3145728
+fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker=RainMaker
+fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker.build.partitions=rainmaker
+fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker.upload.maximum_size=3145728
```
Comment (P-R-O-C-H-Y) [inline boards.txt:26672]: Add new rainmaker partitions and fix the old one.
```suggestion
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker.upload.maximum_size=1966080
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_4MB=RainMaker 4MB No OTA
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_4MB.build.partitions=rainmaker_4MB_no_ota
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_4MB.upload.maximum_size=4038656
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_8MB=RainMaker 8MB
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_8MB.build.partitions=rainmaker_8MB
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_8MB.upload.maximum_size=4116480
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@francisduvivier Last few requests and suggestions, mostly related to the module you use on your board, which is `ESP32-S3-WROOM-1-N16R8` with 16MB flash and 8MB OPI PSRAM. PTAL :)

**Comment by francisduvivier** [comment]
> @francisduvivier Last few requests and suggestions, mostly related to the module you use on your board, which is `ESP32-S3-WROOM-1-N16R8` with 16MB flash and 8MB OPI PSRAM. It's good to remove the options which are not possible to avoid crashing due to wrong selection. PTAL :)

Thanks again for the help! I applied all suggestions in a new commit.

File: boards.txt [hunk 8]
```diff
@@ -26494,6 +26494,219 @@ fm-devkit.menu.EraseFlash.none.upload.erase_cmd=
...
+fri3d_2024_esp32s3.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FATFS)
+fri3d_2024_esp32s3.menu.PartitionScheme.fatflash.build.partitions=ffat
+fri3d_2024_esp32s3.menu.PartitionScheme.fatflash.upload.maximum_size=2097152
+fri3d_2024_esp32s3.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9.9MB FATFS)
+fri3d_2024_esp32s3.menu.PartitionScheme.app3M_fat9M_16MB.build.partitions=app3M_fat9M_16MB
+fri3d_2024_esp32s3.menu.PartitionScheme.app3M_fat9M_16MB.upload.maximum_size=3145728
+fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker=RainMaker
+fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker.build.partitions=rainmaker
+fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker.upload.maximum_size=1966080
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:26663]:
```suggestion
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_4MB=RainMaker 4MB No OTA
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_4MB.build.partitions=rainmaker_4MB_no_ota
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_4MB.upload.maximum_size=4038656
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_8MB=RainMaker 8MB
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_8MB.build.partitions=rainmaker_8MB
fri3d_2024_esp32s3.menu.PartitionScheme.rainmaker_8MB.upload.maximum_size=4116480

```

Comment (francisduvivier) [reply boards.txt:26663]: Sorry bout that, also done now.

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**